### DISCUSSION TITLE
Strange Connectivity problem - pack cli k8s gitlab runner.

### DISCUSSION THREAD

**Discussion by ziadkiwan** [paketo-buildpacks/feedback#26] [Q&A]
Hello Everyone,

We are having a strange problem, and we appreciate any hints!

We have been using Build-pack to build our java mvn apps for multiple years and its we faced never faced any problem!, usually we use runners that are running over docker (docker executors) and over shell (shell executors) lately we started migrating our runners to our K&S cluster that are available on cloud and on premise.

the problem we are facing is that pack CLI is not being able to download the components of the layers (belsoft JDK, Syft....), the runner is able to download the docker runner image and run a docker in docker service (dind)  but when you start downloading the layers to build the image  (belsoft JDK, Syft....) it will just hang, sometime it download the image and verify if and hangs on the next one.
![image](https://user-images.githubusercontent.com/7920707/161248008-9664c67b-a68c-4378-bed5-271a0dee58c1.png)

we tried to skip belsoft JDK but using Orace/Microsoft JDK but then again the download is hanging on syft layer.
![image](https://user-images.githubusercontent.com/7920707/161248232-a36e81c6-f490-41f7-99e4-f3067567463c.png)

if we rollback to docker executor/shell outside of k8s cluster it will work fine the layers will be downloaded successfully.

if we exec on the  gitlab runner container inside the k8s cluster and curl the layers they will download successfully, then if we  docker ps inside the gitlab runner container and docker exec on the docker DIND image and curl the layer component we were able to download the component successfully.
 
![image](https://user-images.githubusercontent.com/7920707/161250797-02067431-70ac-40f1-b36c-845e9eff522b.png)

we even tried on multiple k8s and each k8s have is on a different cluster (linode, AWS, on premise etc) and we are still facing the same issue.
 
 so any hints on what might be the problem? we are lost.

**Comment by dmikusa** [comment]
Happy to help. This forum is used for project feedback/suggestions/feature requests though, not troubleshooting. If you have questions or need technical help, please reach out on [Paketo Slack](https://paketobuildpacks.slack.com
) or open an issue against the problematic buildpack. You're having trouble with Java, so https://github.com/paketo-buildpacks/java/issues would be great. Thanks

**Reply by ziadkiwan** [reply]
Just a side note, would you please grant me access to packeto slack oh email ziad_kiwan_1992@hotmail.com.

**Reply by dmikusa** [reply]
Huh, it should be public and open to everyone. I did send you an invite. Hopefully that'll allow you to get in.

**Comment by ziadkiwan** [comment]
Hello! yes the post here is about java, but we are facing the same issue with all builders, we have java, dotnet and nginx amd all are behaving the same, it was not clear to me where to submit the issue as it seems to be with pack-cli that's why i posted it here, anyway thanks for hints i will check the guys on slack!.
